test/util/util_vault.go

```diff
@@ -0,0 +1,332 @@
+package util
+
+import (
+	"fmt"
+	"os/exec"
+	"path"
+	"time"
+
+	"github.com/golang/glog"
+	vault "github.com/hashicorp/vault/api"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const vaultToken = "vault-root-token"
+
+func NewVaultTokenSecret(name string) *v1.Secret {
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		StringData: map[string]string{
+			"token": vaultToken,
+		},
+	}
+}
+
+func NewVaultAppRoleSecret(name, secretId string) *v1.Secret {
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		StringData: map[string]string{
+			"secretkey": secretId,
+		},
+	}
+}
+
+type VaultInitializer struct {
+	proxyCmd          *exec.Cmd
+	client            *vault.Client
+	rootMount         string
+	intermediateMount string
+	role              string
+}
+
+func NewVaultInitializer(container, rootMount, intermediateMount, role string) (*VaultInitializer, error) {
+	args := []string{"port-forward", "-n", "vault", container, "8200:8200"}
+	cmd := exec.Command("kubectl", args...)
+	err := cmd.Start()
+	if err != nil {
+		glog.Fatalf("Error starting port-forward: %s", err.Error())
+	}
+
+	time.Sleep(3 * time.Second)
+
+	cfg := vault.DefaultConfig()
+	cfg.Address = "http://127.0.0.1:8200"
+	client, err := vault.NewClient(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to initialize vault client: %s", err.Error())
+	}
+
+	client.SetToken(vaultToken)
+
+	return &VaultInitializer{
+		proxyCmd:          cmd,
+		client:            client,
+		rootMount:         rootMount,
+		intermediateMount: intermediateMount,
+		role:              role,
+	}, nil
+}
+
+func (v *VaultInitializer) Setup() error {
```

Comment:
This looks stressful to maintain 😅 

It may be best to move all the vault specific test utils into their own package so you can add an OWNERS file in there to be pinged for code reviews in future.

Reply:
Tests for our tests anyone? 😁 

This can be deferred for now though.